lib/websocket/websocket-handler.ts

```diff
@@ -0,0 +1,238 @@
+socket.log.debug({data: {cid: sub.correlationId, time: ts}}, 'Data received');
+data['DEBUG_TIME'] = ts[0] + '.' + ts[1];
+
+                // Emit the data
+                // TODO: Acknowledgement callback function?
+                socket.emit('data', {'correlationId' : sub.correlationId,
+                                     'data' : data});
+                socket.log.info('Data sent');
+            });
+        });
+
+        // Subscribe user request through blpapi-wrapper
+        socket.blpSession.subscribe(subscriptions)
+        .then((): void => {
+            if (socket.isConnected()) {
+                subscriptions.forEach((s: Subscription): void => {
+                    activeSubscriptions.add(s);
+                });
+                socket.log.debug('Subscribed');
+                socket.emit('subscribed');
+            } else { // Unsubscribe if socket already closed
+                try {
+                    socket.blpSession.unsubscribe(subscriptions);
+                } catch (ex) {
+                    socket.log.error(ex, 'Error Unsubscribing');
+                }
+                subscriptions.forEach((s: Subscription): void => {
+                    s.removeAllListeners();
+                    receivedSubscriptions.delete(s);
+                });
+                socket.log.debug('Unsubscribed all active subscriptions');
+            }
+        })
+        .catch( (err: Error): void => {
+            socket.log.error(err, 'Error Subscribing');
+            subscriptions.forEach((s: Subscription): void => {
+                s.removeAllListeners();
+                receivedSubscriptions.delete(s);
+            });
+            if (socket.isConnected()) {
+                socket.emit('err', err);
+            }
+        });
+    });
+
+    // Unsubscribe
+    socket.on('unsubscribe', (data : { 'correlationIds': number[] }): void => {
+        socket.log.info('Unsubscribe request received');
+
+        if (!activeSubscriptions.size) {
+            socket.log.debug('No active subscriptions.');
+            socket.emit('err', { message: 'No active subscriptions.' });
+            return;
+        }
+
+        var subscriptions : Subscription[] = [];
+
+        // If no correlation Id specified
+        // the default behavior is to unsubscribe all active subscriptions
+        if (!data
+            || !_.has(data, 'correlationIds')
+            || !_.isArray(data.correlationIds)
+            || !data.correlationIds.length) {
+            subscriptions = activeSubscriptions.getAll();
+        }
+        else {
+            // Otherwise, validate all correlation Id
+            // Will unsubscribe only the valid correlation Id, ignore the rest
+            data.correlationIds.forEach((cid: number): void => {
+                if (activeSubscriptions.has(cid)) {
+                    subscriptions.push(activeSubscriptions.get(cid));
+                }
+            });
+        }
+
+        if (!subscriptions.length) {
```

Comment:
My proposal:

* The JSON object must be well-formed (has expected properties), otherwise error
* If the list of `correlationIds` are empty, either unsubscribe all or no-op (we should decide which one).
* In the event there are `correlationIds`
    * If valid, continue
    * If invalid, whether one up to all, ignore